lib/swagger-client.js

```diff
@@ -139,6 +139,8 @@ class Client {
       spec = JSON.parse(zlib.gunzipSync(fs.readFileSync(swaggerPath)))
     }
 
+    const getAliases = alias(options.aliases)
```

Comment:
What if you write your own `getNames` function in https://github.com/nodeshift/openshift-rest-client and we update the `Client` constructor to allow developers to pass in a customer `getNames` function?  That might look something like:

```js
const kubernetesAliases = require('kubernetes-client/alias')
const aliases = {
  'build.openshift.io': ['build']
};

function openshiftAliases(name) {
  if (name in aliases) return aliases.concat([name])
  return kubernetesAliases(name)
}

class OpenshiftClient extends Client {
  constructor(options) {
    return super(Object.assign({ getNames: openshiftAliases }, options))
  }
}
```

Reply:
@silasbw I guess that does look a bit cleaner, if we were able to pass in a `getNames` function.

Do you want me to modify this PR, or should i open a new issue/PR since it is a different thing?